src/main/java/de/wejago/mqtt2influx/service/JsonSubscriber.java

```diff
@@ -65,7 +65,17 @@ private void addMapEntriesByExistingDeviceMappings(Map<String, String> deviceMap
         Map<String, String> messageValueMap = objectMapper.convertValue(entry.getValue(), Map.class);
         for (Map.Entry<String, String> messageValueItem : messageValueMap.entrySet()) {
             if (deviceMappings.containsKey(messageValueItem.getKey())) {
-                deviceToPointProperties.put(deviceMappings.get(messageValueItem.getKey()), messageValueItem.getValue());
+                //we store all numbers as floats in influx
+                if(messageValueItem.getValue().matches("-?\\d+(\\.\\d+)?")) {
+                    try{
+                        deviceToPointProperties.put(deviceMappings.get(messageValueItem.getKey()), Double.parseDouble(messageValueItem.getValue()));
```

Comment:
parseDouble returns a double which is the converted to a Double to be stored in the Map.
Better use valueOf directly.